clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3644,13 +3645,28 @@ static void RenderSSPOptions(const Driver &D, const ToolChain &TC,
           << A->getOption().getName() << Value << "sysreg global";
       return;
     }
+    if (EffectiveTriple.isRISCV()) {
+      if (Value != "tls" && Value != "global") {
```

Comment:
Some of these could probably be simplified if we did something like

```
SmallVector<StringRef, 4> SupportedModes;
if (EffectiveTriple.isX86() || EffectiveTriple.isARM() ||         EffectiveTriple.isThumb() || EffectiveTriple.isRISCV()) {
    SupportedModes = {"global", "tls"};
} else if (EffectiveTriple.isAArch64()) {
    SupportedModes = {"global", "sysreg"};
}
if (!llvm::contains(SupportedModes, Value)) {
      D.Diag(diag::err_drv_invalid_value_with_suggestion)
          << A->getOption().getName() << Value << "sysreg global";
      return;
}
```

Reply:
Yeah, I went for the easiest-to-review mechanism that just created new RISC-V specific clauses instead of attempting to mix them into the existing ones. It's not exactly a performance-critical area of code.

Reply:
Fair enough - this code code do we a bit of refactoring to remove duplication but that doesn't need to be part of this change.